storage/session_redis.go

```diff
@@ -96,7 +97,18 @@ func (s redisSessionStore) Put(key string, value interface{}) error {
 		return err
 	}
 	return s.client.Set(context.Background(), s.toRedisKey(key), marshal, s.ttl).Err()
+}
 
+func (s redisSessionStore) GetAndDelete(key string, target interface{}) error {
+	// GetDel requires redis-server version >= 6.2.0.
+	result, err := s.client.GetDel(context.Background(), s.toRedisKey(key)).Result()
+	if err != nil {
+		if errors.Is(redis.Nil, err) {
+			return ErrNotFound
+		}
+		return err
+	}
+	return json.Unmarshal([]byte(result), target)
```

Comment:
semantics differ from in-memory store; if unmarshal fails here it's still deleted. But I guess that's okay.

Reply:
if unmarshal fails the data is corrupted and therefore unusable. no harm in deleting it